include/libtorrent/aux_/session_impl.hpp

```diff
@@ -1050,21 +1102,43 @@ namespace libtorrent
 			int m_outstanding_router_lookups = 0;
 #endif
 
-			void send_udp_packet_hostname(char const* hostname
+			void send_udp_packet_hostname(std::weak_ptr<utp_socket_interface> sock
+				, char const* hostname
 				, int port
 				, span<char const> p
 				, error_code& ec
 				, int flags);
 
-			void send_udp_packet(bool ssl
+			void send_udp_packet_hostname_listen(aux::session_listen_socket* sock
```

Comment:
I see, that makes sense. This will affect the source port, but not necessarily the source IP, right? because which IP interface to use will still (in the normal case) be determined by the system's routing table, right? I suppose if we have a working bind-to-device, that might not be the case.